CMakeLists.txt

```diff
@@ -9,38 +9,55 @@ include(ExternalProject)
 
 file(MAKE_DIRECTORY ${CATKIN_DEVEL_PREFIX}/include)
 
+SET(CONTRIB_NAME "opencv3_contrib")
+ExternalProject_Add(${CONTRIB_NAME}
+  URL https://github.com/opencv/opencv_contrib/archive/3.1.0.zip
+  UPDATE_COMMAND ""
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ""
+)
+
 ExternalProject_Add(opencv3_src
-  URL https://github.com/Itseez/opencv/archive/3.0.0-beta.zip
+  URL https://github.com/Itseez/opencv/archive/3.1.0.zip
   UPDATE_COMMAND ""
-  PATCH_COMMAND cd ../opencv3_src/
   CONFIGURE_COMMAND cd ../opencv3_src && cmake
-    -D WITH_QT=ON
-    -D WITH_OPENGL=ON
-    -D WITH_VTK=ON
-    -D WITH_OPENMP=ON
-    -D WITH_V4L=ON
-    -D WITH_CUDA=OFF
-    -D WITH_TBB=ON
-    -D BUILD_SHARED_LIBS=ON
-    -D BUILD_EXAMPLES=OFF
-    -D BUILD_PERF_TESTS=OFF
-    -D BUILD_TESTS=OFF
-    -D BUILD_FAT_JAVA_LIB=OFF
-    -D INSTALL_C_EXAMPLES=OFF
-    -D INSTALL_PYTHON_EXAMPLES=OFF
-    -D ENABLE_PRECOMPILED_HEADERS=OFF
-    -D CMAKE_INSTALL_PREFIX=${CATKIN_DEVEL_PREFIX} .
+    -DOPENCV_EXTRA_MODULES_PATH=../../../${CONTRIB_NAME}-prefix/src/${CONTRIB_NAME}/modules .
+    -DWITH_TBB=ON
+    -DBUILD_NEW_PYTHON_SUPPORT=ON
+    -DWITH_V4L=ON
+    -DINSTALL_C_EXAMPLES=OFF
+    -DINSTALL_PYTHON_EXAMPLES=OFF
+    -DBUILD_EXAMPLES=OFF
+    -DWITH_QT=OFF # Needed by clang under Ubuntu 14.04 and GTK_WIDGET(cvGetWindowHandle(...)) with gcc (image_view)
+    -DWITH_OPENGL=ON
+    -DWITH_VTK=ON
+    -DENABLE_PRECOMPILED_HEADERS=OFF
+    -DCMAKE_INSTALL_PREFIX=${CATKIN_DEVEL_PREFIX}
+    -DBUILD_SHARED_LIBS=ON
+    -DWITH_CUDA=OFF
+    -DWITH_OPENCL=OFF
+    -DBUILD_opencv_gpu=OFF
```

Comment:
I can still compile with this (and cuda) turned on. But I didn't test/compile all the way to swe_app (working on muma2 now). I also don't know what happens when you try to compile it on a computer without gpu. It's also disabled in opencv2_catkin